### ISSUE TITLE
Add support for Vim text objects like caq and ciq in Vim mode

### ISSUE THREAD

**Issue by OrlovAlexei** [zed-industries/zed#18295]
### Check for existing issues

- [X] Completed

### Describe the feature

Please add support for the following Vim commands:

caq: Change around quotes — Changes the entire text inside and around quotes, putting the user in insert mode.
ciq: Change inside quotes — Changes only the text inside the quotes, without affecting the quotes themselves.
cab: Change around brackets — Changes the entire text inside and around brackets ( ), { }, or [ ], switching to insert mode.
cib: Change inside brackets — Changes only the text inside the brackets, keeping the brackets intact.
daq / diq: Delete around / inside quotes — Deletes the text around or inside quotes.
dab / dib: Delete around / inside brackets — Deletes text inside or around different types of brackets.

**Comment by ConradIrwin** [comment]
Are you using a plugin for `daq`? We should already support `da"`, and `dab`. I'm not against adding it, but I don't think it's part of vim by default.

The objects we currently support:

https://github.com/zed-industries/zed/blob/44b1399e6c66a5ba4d9a5cdf289458de864a8606/assets/keymaps/vim.json#L373-L396

**Comment by OrlovAlexei** [comment]
You're right, daq is not a default Vim command. The behavior I was referring to typically comes from the popular plugin [vim-surround](https://github.com/tpope/vim-surround) for Vim or [nvim-surround](https://github.com/kylechui/nvim-surround) for Neovim, which adds more flexible and powerful manipulation of surrounding text like quotes, brackets, and other delimiters. 

I was wondering if there's a way to replicate this behavior manually in Zed, or if you have any suggestions on how I could implement something similar. 

Thanks again for your consideration!

**Comment by OrlovAlexei** [comment]
If I want to implement this via keymaps/vim.json, is it possible to add something like vim::AnyQuotes? AnyQuotes would refer to any of the following: Quotes, BackQuotes, or DoubleQuotes. Could something similar be done for brackets as well?

Thanks again for your consideration!

**Comment by oca159** [comment]
Hi @OrlovAlexei and @ConradIrwin,

I’ve created a PR to add support for the AnyQuotes text object: [#22263](https://github.com/zed-industries/zed/pull/22263).

I tested it locally, and it seems to work well, but since I’m not experienced in Rust, I relied heavily on ChatGPT for assistance. 😅 Any advice, suggestions, or feedback to improve this would be greatly appreciated.

Thank you for your time and support!

If you want to test it locally just clone my fork and then run `cargo run` it will create a new instance of zed with the my new code changes. I assigned the new text object to the letter `q`

**Comment by ConradIrwin** [comment]
This was merged!
